src/qtgui/plotter.cpp

```diff
@@ -1187,7 +1180,7 @@ void CPlotter::draw(bool newData)
         && tnow_ms >= tlast_plot_drawn_ms + PLOTTER_UPDATE_LIMIT_MS);
 
     // Do not waste time with histogram calculations unless in this mode.
-    const bool doHistogram = (plotterVisible && m_PlotMode == PLOT_MODE_HISTOGRAM);
+    const bool doHistogram = (plotterVisible && m_PlotMode == PLOT_MODE_HISTOGRAM && newData);
```

Comment:
Processing the histogram after an overlay update causes the IIR data to be blown away.

Reply:
At low frame rates, this make the histogram chase the avg line while panning. I think that's ok, since it also cuts down on processing, since the histogram isn't constantly redrawn during panning. Either way is fine with me.

Reply:
Oops, that was unintended. I was trying to prevent the IIR from being reset when the demod frequency was changed. I'll remove this change and have a look to see whether there's a better way to accomplish that.

Reply:
Actually, it's not that the IIR is reset, just that one IIR averaging step executes each time `updateOverlay` is called, so the IIR rapidly decays even though no new data has arrived.

Reply:
There are some other events that also trigger `updateOverlay` and cause IIR decay:

* filter cutoff adjustment
* changing demod type
* setting A/B markers

Reply:
Changing the condition to `(!m_histIIRValid || newData)` seems to do the trick.

Reply:
Fixed.